test/unit/math/prim/fun/lbeta_test.cpp

```diff
@@ -21,3 +24,79 @@ TEST(MathFunctions, lbeta_nan) {
 
   EXPECT_TRUE(std::isnan(stan::math::lbeta(nan, nan)));
 }
+
+TEST(MathFunctions, lbeta_extremes_errors) {
+  double inf = std::numeric_limits<double>::infinity();
+  double after_stirling
+      = std::nextafter(stan::math::lgamma_stirling_diff_useful, inf);
+  using stan::math::lbeta;
+
+  EXPECT_FLOAT_EQ(lbeta(0.0, 1.0), inf);
+  EXPECT_FLOAT_EQ(lbeta(1.0, 0.0), inf);
+  EXPECT_FLOAT_EQ(lbeta(0.0, after_stirling), inf);
+  EXPECT_FLOAT_EQ(lbeta(after_stirling, 0.0), inf);
+  EXPECT_FLOAT_EQ(lbeta(0.0, 0.0), inf);
+
+  EXPECT_FLOAT_EQ(lbeta(inf, 0.0), inf);
+  EXPECT_FLOAT_EQ(lbeta(0.0, inf), inf);
+  EXPECT_FLOAT_EQ(lbeta(inf, 1), -inf);
+  EXPECT_FLOAT_EQ(lbeta(1e8, inf), -inf);
+  EXPECT_FLOAT_EQ(lbeta(inf, inf), -inf);
+}
+
+TEST(MathFunctions, lbeta_identities) {
+  using stan::math::lbeta;
+  using stan::math::pi;
+  using stan::test::expect_near_rel;
+
+  std::vector<double> to_test
+      = {1e-100, 1e-8, 1e-1, 1, 1 + 1e-6, 1e3, 1e30, 1e100};
+  for (double x : to_test) {
+    for (double y : to_test) {
+      std::ostringstream msg;
+      msg << "successors: x = " << x << "; y = " << y;
+      expect_near_rel(
+          msg.str(), lbeta(x, y),
+          stan::math::log_sum_exp(lbeta(x + 1, y), lbeta(x, y + 1)));
+    }
+  }
+
+  for (double x : to_test) {
+    if (x < 1) {
+      std::ostringstream msg;
+      msg << "sin: x = " << x;
+      expect_near_rel(msg.str(), lbeta(x, 1.0 - x),
+                      log(pi()) - log(sin(pi() * x)));
+    }
+  }
+
+  for (double x : to_test) {
+    std::ostringstream msg;
+    msg << "inv: x = " << x;
+    expect_near_rel(msg.str(), lbeta(x, 1.0), -log(x));
+  }
+}
+
+TEST(MathFunctions, lbeta_stirling_cutoff) {
+  using stan::test::expect_near_rel;
+
+  double after_stirling
+      = std::nextafter(stan::math::lgamma_stirling_diff_useful,
```

Comment:
I would put a `using stan::math::lgamma_stirling_diff_useful` since it's used a few times and lines get very long.